src/main/java/htsjdk/samtools/util/SnappyLoader.java

```diff
@@ -38,73 +39,71 @@
+            snappyAvailable = false;
         }
         else {
             boolean tmpSnappyAvailable = false;
-            try {
-                final OutputStream test = new SnappyOutputStream(new ByteArrayOutputStream(1000));
+            try (final OutputStream test = new SnappyOutputStream(new ByteArrayOutputStream(1000))){
                 test.write("Hello World!".getBytes());
-                test.close();
                 tmpSnappyAvailable = true;
-                if (verbose) logger.info("Snappy successfully loaded.");
+                logger.debug("Snappy successfully loaded.");
             }
             /*
              * ExceptionInInitializerError: thrown by Snappy if native libs fail to load.
              * IllegalStateException: thrown within the `test.write` call above if no UTF-8 encoder is found.
              * IOException: potentially thrown by the `test.write` and `test.close` calls.
              * SnappyError: potentially thrown for a variety of reasons by Snappy.
              */
-            catch (ExceptionInInitializerError|IllegalStateException|IOException|SnappyError e) {
-                if (verbose) logger.warn("Snappy native library failed to load: " + e.getMessage());
+            catch (final ExceptionInInitializerError | IllegalStateException | IOException | SnappyError e) {
+                logger.warn(e, "Snappy native library failed to load.");
             }
-            SnappyAvailable = tmpSnappyAvailable;
+            snappyAvailable = tmpSnappyAvailable;
         }
     }
 
     /** Returns true if Snappy is available, false otherwise. */
-    public boolean isSnappyAvailable() { return SnappyAvailable; }
+    public boolean isSnappyAvailable() { return snappyAvailable; }
 
-    /** Wrap an InputStream in a SnappyInputStream. If Snappy is not available will throw an exception. */
+    /**
+     * Wrap an InputStream in a SnappyInputStream.
+     * @throws SAMException if Snappy is not available will throw an exception.
+     */
     public InputStream wrapInputStream(final InputStream inputStream) {
-        if (isSnappyAvailable()) {
-            try {
-                return new SnappyInputStream(inputStream);
-            } catch (Exception e) {
-                throw new SAMException("Error instantiating SnappyInputStream", e);
-            }
-        } else {
-            throw new SAMException("Snappy not available");
-        }
+        return wrapWithSnappyOrThrow(inputStream, SnappyInputStream::new);
     }
 
-    /** Wrap an InputStream in a SnappyInputStream. If Snappy is not available will throw an exception. */
+    /**
+     * Wrap an OutputStream in a SnappyOutputStream.
+     * @throws SAMException if Snappy is not available
+     */
     public OutputStream wrapOutputStream(final OutputStream outputStream) {
+        return wrapWithSnappyOrThrow(outputStream, (stream) -> new SnappyOutputStream(stream, SNAPPY_BLOCK_SIZE));
+    }
+
+    private interface IOFunction<T,R> {
```

Comment:
I'm not really up on the Java 8 lambda stuff, but is this interface needed?  Can't you just use `java.util.Function<T,R>' in it's place?

Reply:
sadly you can't, the default types don't throw checked exceptions, so you can't capture something that throws IOException as a standard `Function` there doesn't yet seem to a be standard set of "throwing interfaces" anywhere yet either.